app/Http/Controllers/Auth/VATSIMOAuthController.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Http\Controllers\Auth;
+
+use League\OAuth2\Client\Provider\Exception\IdentityProviderException;
+use League\OAuth2\Client\Provider\GenericProvider;
+
+class VATSIMOAuthController extends GenericProvider
+{
+    /**
+     * @var GenericProvider
+     */
+    private $provider;
+
+    /**
+     * Initializes the provider variable.
+     */
+    public function __construct()
+    {
+        parent::__construct([
+            'clientId'                => config('vatsim-connect.id'),
+            'clientSecret'            => config('vatsim-connect.secret'),
+            'redirectUri'             => route('login.post'),
+            'urlAuthorize'            => config('vatsim-connect.base').'/oauth/authorize',
+            'urlAccessToken'          => config('vatsim-connect.base').'/oauth/token',
+            'urlResourceOwnerDetails' => config('vatsim-connect.base').'/api/user',
+            'scopes'                  => config('vatsim-connect.scopes'),
+            'scopeSeparator'          => ' ',
+        ]);
+    }
+
+    /**
+     * Gets an (updated) user token.
+     * @param Token $token
+     * @return Token
+     * @return null
+     */
+    public static function updateToken($token)
+    {
+        $controller = new self;
+        try {
+            return $controller->getAccessToken('refresh_token', [
+                'refresh_token' => $token->getRefreshToken(),
+            ]);
+        } catch (IdentityProviderException $e) {
+            return;
```

Comment:
Is there a better course of action here rather than just exiting the function? Or, should this exception be caught by consumers of the function?

Reply:
I would presume that this is there to catch the exception that will be thrown if the user has revoked our access to their account on VATSIM's side.